solr/core/src/java/org/apache/solr/handler/admin/ZookeeperStatusHandler.java

```diff
@@ -112,15 +112,21 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
           .map(h -> h.resolveClientPortAddress() + ":" + h.clientPort)
           .sorted().collect(Collectors.toList());
       List<String> dynamicHosts = zkDynamicConfig.getServers().stream()
-          .map(h -> h.resolveClientPortAddress() + ":" + h.clientPort)
+          .map(h -> h.resolveClientPortAddress() + ":" + (h.clientPort != null ? h.clientPort : "2181"))
```

Comment:
This is not bullet proof if e.g. user has `clientPort=1234` in `zoo.cfg` and in zkHost connection string. Then we'll add a warning that dynamic config differs from zkHost, which is not entirely true since we just lack the port part. We have no way from client to read the `clientPort` from server except from connecting to the server with 4LW *ont the clientPort* which is a chicken and egg. This hack will make the comparison work for default port, which is a compromise.

Reply:
Maybe instead of defaulting to `2181`, we default to `hostsFromConnectionString.getServers().get(0).clientPort`? I imagine that would match a vast majority of scenarios.

Reply:
Agree, I'll do that change and merge